validator/keymanager/remote-web3signer/internal/client.go

```diff
@@ -116,27 +120,35 @@ func (client *ApiClient) GetServerStatus(ctx context.Context) (string, error) {
 
 // doRequest is a utility method for requests.
 func (client *ApiClient) doRequest(ctx context.Context, httpMethod, fullPath string, body io.Reader) (*http.Response, error) {
+	ctx, span := trace.StartSpan(ctx, "remote_web3signer.client.doRequest")
+	defer span.End()
+	span.AddAttributes(
+		trace.StringAttribute("httpMethod", httpMethod),
+		trace.StringAttribute("fullPath", fullPath),
+		trace.StringAttribute("hasBody", fmt.Sprintf("%v", body != nil)),
+	)
 	req, err := http.NewRequestWithContext(ctx, httpMethod, fullPath, body)
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid format, failed to create new Post Request Object")
 	}
 	req.Header.Set("Content-Type", "application/json")
+	requestDump, err := httputil.DumpRequest(req, true)
+	if err != nil {
+		return nil, err
+	}
+	start := time.Now()
 	resp, err := client.RestClient.Do(req)
+	duration := time.Since(start)
 	if err != nil {
+		signRequestDurationSeconds.WithLabelValues(req.Method, "error").Observe(duration.Seconds())
 		return resp, errors.Wrap(err, "failed to execute json request")
```

Comment:
Another thing you can do is to annotate the span with the error. 

https://github.com/prysmaticlabs/prysm/blob/9935ca3733b33eb1d8b28d86c75062316090bd24/monitoring/tracing/errors.go#L9

```suggestion
	if err != nil {
		signRequestDurationSeconds.WithLabelValues(req.Method, "error").Observe(duration.Seconds())
		err = errors.Wrap(err, "failed to execute json request")
		tracing.AnnotateError(span, err)
		return resp, err
```